File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        l, r = 0, len(height) - 1
        maxAmount=0
        
        while l < r:
            cur_width = r - l
            cur_height = min(height[l], height[r])
            curAmount = cur_width * cur_height
            if curAmount > maxAmount:
                maxAmount = curAmount
            if height[l] > height[r]:
                r-=1
            else:
                l+=1
        
        return maxAmount
            
            
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int cur = 0, max = 0;
        int l = 0, r = height.size() - 1;

        while (l < r) {
            height[l] > height[r] ? cur = height[r] * (r - l ) : cur = height[l] * (r - l );
            if (cur > max) max = cur;
            height[l] > height[r] ? r-- : l++;
        }

        return max;

    }
};